libraries/bxdf/gltf_pbr.mtlx

```diff
@@ -266,4 +266,477 @@
+    <input name="file" type="filename" uniform="true" value="" />
+    <input name="factor" type="color3" value="1,1,1" />
+    <input name="default" type="color3" value="0, 0, 0" />
+    <input name="uvindex" type="integer" value="0" />
+    <input name="pivot" type="vector2" value="0, 1" />
+    <input name="scale" type="vector2" value="1, 1" />
+    <input name="rotate" type="float" value="0" unit="degree" unittype="angle" uimin="0" uimax="360" />
+    <input name="offset" type="vector2" value="0, 0" />
+    <input name="operationorder" type="integer" value="0" />
+    <input name="uaddressmode" type="string" uniform="true" value="periodic" />
+    <input name="vaddressmode" type="string" uniform="true" value="periodic" />
+    <input name="filtertype" type="string" uniform="true" value="linear" />
+    <output name="out" type="color3" value="0, 0, 0" />
+  </nodedef>
+  <nodegraph name="NG_NG_gltf_image_color3_color3_1_0" nodedef="ND_gltf_image_color3_color3_1_0">
+    <input name="file" type="filename" uniform="true" value="" />
+    <input name="factor" type="color3" value="1,1,1" />
+    <input name="default" type="color3" value="0, 0, 0" />
+    <input name="uvindex" type="integer" value="0" />
+    <input name="pivot" type="vector2" value="0, 1" />
+    <input name="scale" type="vector2" value="1, 1" />
+    <input name="rotate" type="float" value="0" />
+    <input name="offset" type="vector2" value="0, 0" />
+    <input name="operationorder" type="integer" value="0" />
+    <input name="uaddressmode" type="string" uniform="true" value="periodic" />
+    <input name="vaddressmode" type="string" uniform="true" value="periodic" />
+    <input name="filtertype" type="string" uniform="true" value="linear" />
+    <texcoord name="texcoord1" type="vector2">
+      <input name="index" type="integer" uniform="true" interfacename="uvindex" />
+    </texcoord>
+    <image name="image" type="color3">
+      <input name="file" type="filename" uniform="true" interfacename="file" />
+      <input name="default" type="color3" interfacename="default" value="0, 0, 0" />
+      <input name="texcoord" type="vector2" nodename="place2d" />
+      <input name="uaddressmode" type="string" uniform="true" interfacename="uaddressmode" value="periodic" />
+      <input name="vaddressmode" type="string" uniform="true" interfacename="vaddressmode" value="periodic" />
+      <input name="filtertype" type="string" uniform="true" interfacename="filtertype" value="linear" />
+    </image>
+    <divide name="invert_scale" type="vector2">
+      <input name="in1" type="vector2" value="1.0, 1.0"/>
+      <input name="in2" type="vector2" interfacename="scale"/>
+    </divide>
+    <multiply name="negate_rotate" type="float">
+      <input name="in1" type="float" interfacename="rotate"/>
+      <input name="in2" type="float" value="-1.0"/>
+    </multiply>
+    <multiply name="negate_offset" type="vector2">
```

Comment:
Leaving a note here. One reason why the glTF / MTLX web viewer is inefficient is that the actual geometry is being remapped to another set of streams to support the MTLX lower-left convention for offset. It would be good to allow for a offset origin. This also applies to rotation and scale as
this is using TRS vs SRT transform convention (which was recently added).